lua/hbac/command/subcommands.lua

```diff
@@ -0,0 +1,45 @@
+local autocommands = require("hbac.autocommands")
+local hbac_config = require("hbac.setup").opts
+local state = require("hbac.state")
+local utils = require("hbac.utils")
+
+local M = {}
+
+M.close_unpinned = function()
+	local curbufnr = vim.api.nvim_get_current_buf()
+	local buflist = utils.get_buffers()
+	for _, bufnr in ipairs(buflist) do
+		local windows_with_buf = vim.fn.win_findbuf(bufnr)
+		if bufnr ~= curbufnr and not state.is_pinned(bufnr) and #windows_with_buf == 0 then
+			hbac_config.close_command(bufnr)
+		end
+	end
+end
+
+M.toggle_pin = function()
+	local bufnr = vim.api.nvim_get_current_buf()
+	local pinned_state = state.toggle_pin(bufnr) and "pinned" or "unpinned"
+	return bufnr, pinned_state
+end
+
+M.pin_all = function()
```

Comment:
I guess this is toggle_all because it works both ways, but I'm ok with pin_all as well

Reply:
Yes my mistake, I forgot to change `toggle_all` to `pin_all` in the subcommands module. Both names are kind of misleading though (pin_all doesn't invite the user to try to unpin all and toggle_all suggests toggling the state of all buffers to their opposite rather than all to pinned/unpinned). Naming things is hard.